packages/client/src/client/open-feature-client.ts

```diff
@@ -225,6 +226,19 @@ export class OpenFeatureClient implements Client {
         flagKey,
       };
 
+      if (evaluationDetails.errorCode) {
+        this.errorHooks(
```

Comment:
In the Python SDK we construct and raise an appropriate exception here to avoid duplicating the error handling code from the catch clause. Would it make sense to do something similar?

Reply:
I did a little investigation into this and it looks like in node, stack traces are captured in the error constructor. So calling `new Error()` captures the stack trace and results in significant performance penalty ([reference](https://nodejs.org/api/errors.html#class-error)). This is news to me, I actually thought it happened with the throw (something c# does).

Provider authors returning an error the way we do in this PR has some benefits, in that if it's a error condition that's well understood and doesn't benefit from a stack trace and a proper exception, they could return an error and avoid the performance hit. If we instantiate our own exception anyway, we've negated that performance improvement.

We _could_ instantiate and throw a non-error, but that could be confusing and lead to unexpected bugs.

Reply:
But that's what `instantiateErrorByErrorCode` does a couple of lines below, or am I missing something? If I understand you correctly, the performance cost of instantiating this error would be paid in both cases.

Reply:
Ya, I was being very unclear.

This PR _does_ carry the performance penalty I mentioned already. The way to avoid that would be to construct something shaped like an error that doesn't capture the stack trace, that's what I meant by:

> We could instantiate and throw a non-error, but that could be confusing and lead to unexpected bugs.

Irrespective of that, the flow could be improved by doing what you suggest I think; sorry, I got carried away by my own research how the stack trace is built and got away from your original point.

Reply:
I've made this change. Thanks for the suggestion.